hw/ip/otbn/dv/otbnsim/sim/csr.py

```diff
@@ -36,19 +47,23 @@ def read_unsigned(self, wsrs: WSRFile, idx: int) -> int:
     def write_unsigned(self, wsrs: WSRFile, idx: int, value: int) -> None:
         assert 0 <= value < (1 << 32)
 
-        if idx == 0x7c0:
+        if 0x7c0 <= idx <= 0x7c1:
+            # FG0/FG1
+            self.flags.write_unsigned(
+                self._write_field(idx - 0x7c0, 4, value,
+                                  self.flags.read_unsigned()))
```

Comment:
I'm not a huge fan of this sort of indentation. Maybe use some variables to give names to some values (for example, `idx - 0x7c0` might not be obvious to a reader) and split up the big expression. How about this?
```python
        if 0x7c0 <= idx <= 0x7c1:
            # FG0/FG1
            fg = idx - 0x7c1
            old = self.flags.read_unsigned()
            self.flags.write_unsigned(self._write_field(fg, 4, value, old))
            return
```
and similarly for MOD.